api/applicationserver_packages.proto

```diff
@@ -0,0 +1,107 @@
+import "github.com/gogo/protobuf/gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/struct.proto";
+import "lorawan-stack/api/identifiers.proto";
+
+package ttn.lorawan.v3;
+
+option go_package = "go.thethings.network/lorawan-stack/pkg/ttnpb";
+
+message ApplicationPackage {
+  string name = 1 [(validate.rules).string = {pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$" , max_len: 36}];
+  uint32 default_f_port = 2 [(validate.rules).uint32 = {gte: 1, lte: 255}];
+}
+
+message ApplicationPackages {
+  repeated ApplicationPackage packages = 1;
+}
+
+message ApplicationPackageAssociationIdentifiers {
+  EndDeviceIdentifiers end_device_ids = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true, (validate.rules).message.required = true];
+  uint32 f_port = 2 [(validate.rules).uint32 = {gte: 1, lte: 255}];
+}
+
+message ApplicationPackageAssociation {
+  ApplicationPackageAssociationIdentifiers ids = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true, (validate.rules).message.required = true];
+  google.protobuf.Timestamp created_at = 2 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+  google.protobuf.Timestamp updated_at = 3 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
+  string package_name = 4 [(validate.rules).string = {pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$" , max_len: 20}];
+  google.protobuf.Struct data = 5;
+}
+
+message ApplicationPackageAssociations {
+  repeated ApplicationPackageAssociation associations = 1;
+}
+
+message GetApplicationPackageAssociationRequest {
+  ApplicationPackageAssociationIdentifiers ids = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true, (validate.rules).message.required = true];
+  google.protobuf.FieldMask field_mask = 2 [(gogoproto.nullable) = false];
+}
+
+message ListApplicationPackageAssociationRequest {
+  EndDeviceIdentifiers ids = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true, (validate.rules).message.required = true];
+  google.protobuf.FieldMask field_mask = 2 [(gogoproto.nullable) = false];
+}
+
+message SetApplicationPackageAssociationRequest {
+  ApplicationPackageAssociation association = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true, (validate.rules).message.required = true];
+  google.protobuf.FieldMask field_mask = 2 [(gogoproto.nullable) = false];
+}
+
+service ApplicationPackageRegistry {
+  // GetPackages returns the available packages for the end device.
+  rpc GetPackages(EndDeviceIdentifiers) returns (ApplicationPackages) {
+    option (google.api.http) = {
+      get: "/as/applications/{application_ids.application_id}/devices/{device_id}/packages"
+    };
+  }
+
+  // Get returns the application package association registered on the FPort of the end device.
+  rpc Get(GetApplicationPackageAssociationRequest) returns (ApplicationPackageAssociation) {
```

Comment:
This would be `GetAssociation` (the `Get` of the `ApplicationPackageRegistry` would get a `Package`)